presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -128,6 +129,24 @@ else if (simplified.equals(FALSE_LITERAL) || simplified instanceof NullLiteral)
         }
 
         @Override
+        public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
```

Comment:
o_O I did not know that `SimplifyExpression` was missing a support for join. Great find!

@szymonm Just take a look your `ExpressionRewriteRuleSet` would just solve it automatically once `SimplifyExpression` got migrated.